src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -607,32 +611,27 @@ static int guigetstr(Object* ho, char** cpp) {
     return 1;
 }
 
-static PyObject* loads;
-static PyObject* dumps;
+static nb::callable loads;
+static nb::callable dumps;
 
 static void setpickle() {
     if (dumps) {
         return;
     }
-    PyObject* pickle = PyImport_ImportModule("pickle");
-    if (pickle) {
-        Py_INCREF(pickle);
-        dumps = PyObject_GetAttrString(pickle, "dumps");
-        loads = PyObject_GetAttrString(pickle, "loads");
-        if (dumps) {
-            Py_INCREF(dumps);
-            Py_INCREF(loads);
-        }
-    }
+    nb::module_ pickle = nb::module_::import_("pickle");
+    dumps = pickle.attr("dumps");
+    loads = pickle.attr("loads");
     if (!dumps || !loads) {
         hoc_execerror("Neither Python cPickle nor pickle are available", 0);
     }
+    dumps.inc_ref();
+    loads.inc_ref();
```

Comment:
I don't understand why this is needed.

Reply:
But we don't care because this is static so it can leak.

Reply:
I suspect that it's because the dtor of these will run after Python has shut down. It's plausible that the `DECREF` code to decrement the counter doesn't depend on anything that's been deallocated, while the code to clean up *if* one hits refcount `0` requires something that's not available anymore.

You're right, these can leak.